repository_browser/lib/main.dart

```diff
@@ -64,17 +61,43 @@ class MyApp extends StatelessWidget {
 class DetailScreen extends StatelessWidget {
   final String commitsUrl;
   DetailScreen(this.commitsUrl);
-  
+
+  Future <List<String>> searchCommitsMessage() async {
+    print(' start !!! ${commitsUrl}');
+    final url = commitsUrl.substring(0, commitsUrl.indexOf('{'));
```

Comment:
Imagine that someone else is reading your code. They will not understand why you need substring. Please leave a comment to explain what you are doing with substring.